packages/crawler/src/page-processors/page-processor-factory.ts

```diff
@@ -0,0 +1,7 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+import { PageProcessor, PageProcessorOptions } from './page-processor-base';
+
+export interface PageProcessorFactory {
```

Comment:
why we are moving away from objects to functional programming? we can have interface and abstract class at the same time. not clear why need to create get-discovery-pattern.ts with functions and drop abstract class.

Reply:
This was another thing I moved for ease of testing, but since without this function the PageProcessorFactories are so simple that they don't necessarily need their own tests, I guess it isn't needed.